src/transformers/pipelines/image_segmentation.py

```diff
@@ -0,0 +1,150 @@
+                # like http_huggingface_co.png
+                return Image.open(requests.get(image, stream=True).raw)
+            elif os.path.isfile(image):
+                return Image.open(image)
+        elif isinstance(image, Image.Image):
+            return image
+
+        raise ValueError(
+            "Incorrect format used for image. Should be an url linking to an image, a local path, or a PIL image."
+        )
+
+    def __call__(
+        self,
+        images: Union[str, List[str], "Image", List["Image"]],
+        threshold: Optional[float] = 0.9,
+        mask_threshold: Optional[float] = 0.9,
+    ) -> Union[Predictions, List[Prediction]]:
+        """
+        Assign labels to the image(s) passed as inputs.
+
+        Args:
+            images (:obj:`str`, :obj:`List[str]`, :obj:`PIL.Image` or :obj:`List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images, which must then be passed as a string.
+                Images in a batch must all be in the same format: all as http links, all as local paths, or all as PIL
+                images.
+            threshold (:obj:`float`, `optional`, defaults to 0.9):
+                The probability necessary to make a prediction.
+            mask_threshold (:obj:`float`, `optional`, defaults to 0.9):
+                The probability necessary to keep the pixel within the mask of said prediction.
+
+        Return:
+            A list of dictionaries or a list of of list of dictionaries containing result. If the input is a single
+            image, will return a list of dictionaries, if the input is a list of several images, will return a list of
+            list of dictionaries corresponding to each image.
+
+            The dictionaries contain the following keys:
+
+            - **label** (:obj:`str`) -- The label identified by the model. There could be duplicates within the list
+              (corresponding to instances within a panoptic segmentation.
+            - **score** (:obj:`int`) -- The score attributed by the model for that label as a probability.
+            - **mask** (:obj:`np.array`) -- The bitmask of shape (H, W) of the input image, 0 means the class is not
+              affected to the label, 1 means it is.
+        """
+        is_batched = isinstance(images, list)
+
+        if not is_batched:
+            images = [images]
+
+        images = [self.load_image(image) for image in images]
+
+        with torch.no_grad():
+            inputs = self.feature_extractor(images=images, return_tensors="pt")
+            outputs = self.model(**inputs)
+
+            processed_sizes = [(im.height, im.width) for im in images]
```

Comment:
In case images are PIL images, then width comes first, so it should be (im.width, im.height).

For (PyTorch) tensors, height comes first.